core/dbt/task/init.py

```diff
@@ -274,9 +276,18 @@ def run(self):
             in_project = False
 
         if in_project:
+            # If --profile was specified, it means use an existing profile, which is not
+            # applicable to this case
+            if getattr(get_flags(), "PROFILE", None):
+                print(
```

Comment:
Let's import and use an appropriate exception from `dbt.exceptions` which is defined in `core/dbt/exceptions`. Maybe `DbtRuntimeError`?